src/dag-node/toDagLink.js

```diff
@@ -1,14 +1,15 @@
 'use strict'
 
-const DAGLink = require('./../dag-link/dagLink')
-const { cid, serialize } = require('../util')
+const DAGLink = require('../dag-link/dagLink')
+const { serializeDAGNode } = require('../serialize')
+const genCid = require('../genCid')
 
 /*
  * toDAGLink converts a DAGNode to a DAGLink
  */
 const toDAGLink = async (node, options = {}) => {
-  const serialized = serialize(node)
-  const nodeCid = await cid(serialized)
+  const serialized = serializeDAGNode(node)
+  const nodeCid = await genCid.cid(serialized)
```

Comment:
It'd be great if we could avoid the double-serialization thing.

The typical usecase is having just `ipld.put`'d a node, you then want to add it as a link to a parent node, for which you need the `cid` and the serialized size.

The `cid` is returned by `ipld.put` but the serialized size is not available.  I suppose you could ask the block store directly, but it'd need some sort of lru cache so it wouldn't go to disk for every request.

Better yet would be to remove the idea of sizes from `DAGLink`s as they are unverifiable anyway.

Reply:
> Better yet would be to remove the idea of sizes from `DAGLink`s as they are unverifiable anyway.

Is the `Tsize` in the `PBLink` the same as the `DAGLink` size? If yes, would it mean that you would need to change UnixFS in order to get rid of the `DAGLink` size?

Can this problem be discussed outside this PR? I'd really like to move the dag-pb API changes forward.

Reply:
> Would it mean that you would need to change UnixFS in order to get rid of the DAGLink size?

Yes, I believe so though I think it's no bad thing - we don't use that field because the useful thing is the size of the file, not the size of the DAG that holds the file.

> Can this problem be discussed outside this PR?

Absolutely, I didn't mean to suggest we should resolve that here.